In [2]:
import pandas as pd
import numpy as np
import math

#### Задание-1

In [3]:
# Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].
# Найдите ее среднее значение и дисперсию.

In [4]:
a = 200
b = 800
M = (a + b) / 2
M

500.0

In [5]:
D = ((b - a) ** 2) / 12
D

30000.0

*Ответ: дисперсия равна 30000, а среднее значение равно 500.*

#### Задание-2

In [6]:
# О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.
# Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5?
# Если да, найдите ее.

In [7]:
# Решение: Величина В распределена на отрезхке ab, где D=0.2, a=0.5.
# а) Найти b. 
# б) Найти M.

In [8]:
# a):
# D = ((b - a) ** 2) / 12
# Подставляем в формулу значение a и получаем квадратное уравнение:
# 0.2 = (b ** 2 - 2 * 0.5 * b + 0.5 ** 2) / 12
# b ** 2 - b - 2.15 = 0 

In [9]:
print("Введите коэффициенты для уравнения")
print("ax^2 + bx + c = 0:")
# a = float(input("a = "))
# b = float(input("b = "))
# c = float(input("c = "))

a = 1
b = -1
c = -2.15
 
discr = b ** 2 - 4 * a * c
print("Дискриминант D = %.2f" % discr)
 
if discr > 0:
    x1 = (-b + math.sqrt(discr)) / (2 * a)
    x2 = (-b - math.sqrt(discr)) / (2 * a)
    print("x1 = %.2f \nx2 = %.2f" % (x1, x2))
elif discr == 0:
    x = -b / (2 * a)
    print("x = %.2f" % x)
else:
    print("Корней нет")

Введите коэффициенты для уравнения
ax^2 + bx + c = 0:
Дискриминант D = 9.60
x1 = 2.05 
x2 = -1.05


In [10]:
# Так как b по условию задачи является правой границей,
# то b=2.05

*Ответ: b примерно равно 2.05.*

In [11]:
# b):
# M = (a + b) / 2

In [12]:
a = 0.5
b = 2.05
M = (a + b) / 2
M

1.275

*Ответ: М равно 1.275 (1.2746).*

#### Задание-3

In [13]:
# Непрерывная случайная величина X распределена нормально и задана плотностью распределения
# f(x) = (1 / (4 * sqrt(2*pi))) * (exp(-(x+2)**2) / 32).
# Найдите:
# а). M(X)
# б). D(X)
# в). std(X) (среднее квадратичное) 

In [14]:
# Обратимся к формуле распределения плотности вероятностей непрерывной 
# случайной величины Гаусса: f(x)=(1/(sig*sqrt(2*pi)))*(exp(-(x-a)**2) / (2*(sig**2)),
# где: a=M(X), sig**2=D(X)
# Из условия задачи получаем:
# а) M(X) = -2
# б) D(X) = 16 
# в) std(X) = 4

*Ответ: М(Х)=-2, D(X)=16, std(X)=4.*

#### Задание-4

In [15]:
# Рост взрослого населения города X имеет нормальное распределение.
# Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.
# Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
# а). больше 182 см
# б). больше 190 см
# в). от 166 см до 190 см
# г). от 166 см до 182 см
# д). от 158 см до 190 см
# е). не выше 150 см или не ниже 190 см
# ё). не выше 150 см или не ниже 198 см
# ж). ниже 166 см.

In [16]:
# Решение: чтобы найти вероятность Р необходимо обратиться к таблицам накопленного
# нормального распределения N(Z), где Z=(X-mu)/sig.
mu = 174
sig = 8
def z_h(x):
    return ((x - mu) / sig)

In [17]:
# а). больше 182 см
z_h(182)

1.0

In [18]:
P = (1 - 0.84134) * 100
P

15.866000000000003

*Ответ а): вероятность 15.866 процентов.*

In [19]:
# б). больше 190 см
z_h(190)

2.0

In [20]:
P = (1 - 0.97725) * 100
P

2.275000000000005

*Ответ б): вероятность 2.275 проценттов.*

In [21]:
# в). от 166 см до 190 см
z_h(166)

-1.0

In [22]:
# P(166) = 15.866 процентов 

In [23]:
z_h(190)

2.0

In [24]:
# P(190) = 97.725 процентов.

In [25]:
# P = p(190) - p(166)
P = 97.725 - 15.866
P

81.859

*Ответ в): вероятность 81.859 процентов.*

In [26]:
# г). от 166 см до 182 см
P = 84.134 - 15.866
P

68.268

*Ответ г): вероятность 62.268 процентов.*

In [27]:
# д). от 158 см до 190 см
z_h(158)

-2.0

In [28]:
# P(158) = 2.275 процента
# P = P(190) - P(158)
P = 97.725 - 2.275
P

95.44999999999999

*Ответ д): вероятность 95.449 процентов.*

In [29]:
# е). не выше 150 см или не ниже 190 см
z_h(150)

-3.0

In [30]:
# P(150) = 0.135 процента
# P = P(150) + (1 - P(190))
P = 0.135 + 2.275
P

2.41

*Ответ е): вероятность 2.41 процента.*

In [31]:
# ё). не выше 150 см или не ниже 198 см
z_h(198)

3.0

In [32]:
# P(198) = 0.135 процента
# P = P(150) + (1 - P(198))
P = 0.135 + 0.135
P

0.27

*Ответ ё): вероятность 0.27 процента.*

In [33]:
# ж). ниже 166 см.
# P(166) = 15.866 процентов.

*Ответ ж): вероятность 15.866 процентов.*

#### Задание-5

In [34]:
# На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, 
# равный 190 см, от математического ожидания роста в популяции, 
# в которой M(X) = 178 см и D(X) = 25 кв.см?

In [35]:
# Решение: в соответствии с формулой Z=(X-mu)/sig получаем:
Z = (190 - 178) / math.sqrt(25)
Z

2.4

*Ответ: рост человека в 190 см отклоняется от матожидания на +2.4 сигмы.*